src/include/txn_inline.h

```diff
@@ -1194,6 +1194,15 @@ __wt_txn_id_check(WT_SESSION_IMPL *session)
     if (F_ISSET(txn, WT_TXN_HAS_ID))
         return (0);
 
+    /*
+     * Return error when the transactions with read committed or uncommitted isolation tries to
+     * perform any write operation.
+     */
+    if (session->dhandle != NULL && !WT_IS_METADATA(session->dhandle) &&
```

Comment:
Alex has mentioned as a comment on WT-6677 a few extra things that can be done. Adding a comment around metadata exception and adding an assertion for internal threads. I will wait for those before I review again.

Reply:
@sulabhM Updated the comments and added an assert for internal threads.